lib/logstorage/block.go

```diff
@@ -41,42 +41,43 @@ func (b *block) reset() {
 
 // uncompressedSizeBytes returns the total size of the original log entries stored in b.
 //
-// It is supposed that every log entry has the following format:
-//
-// 2006-01-02T15:04:05.999999999Z07:00 field1=value1 ... fieldN=valueN
-func (b *block) uncompressedSizeBytes() uint64 {
-	rowsCount := uint64(b.Len())
-
-	// Take into account timestamps
-	n := rowsCount * uint64(len(time.RFC3339Nano))
+// It uses JSON format to calculate the size as if each log entry were represented as JSON.
+func (b *block) uncompressedSizeBytes() int {
+	rowsCount := b.Len()
+	if rowsCount == 0 {
+		return 0
+	}
+
+	totalSize := (len("{}\n") + len(`"_time":""`) + len(time.RFC3339Nano)) * rowsCount
+
+	// size of constant fields (included in every row)
+	constFields := make([]Field, len(b.constColumns))
```

Comment:
Why the `constFields` is needed here? This is a memory allocation in a hot path (the `uncompressedSizeBytes()` is called per each written data block inside the `block.mustWriteTo()`). This may significantly slow down this hot path. Memory allocations must be avoided in hot paths.